[Jules]: fix typescript build issues again, again

### Task Requirements

You are a Senior Frontend Architect. Your task is to resolve the remaining TypeScript compilation errors (TS2339 and TS2307) preventing the production build of the Meraki HA panel.

<detected_errors>
1. TS2339: Property 'ha-...' does not exist on type 'JSX.IntrinsicElements'.
2. TS2307: Cannot find module './types/websocket' or its corresponding type declarations.
</detected_errors>

<instructions>
1. Check the project requirements to prevent regressions and update the requirements document based on this fix.
2. Think step-by-step. We must explicitly register the Home Assistant Web Components in the global JSX namespace so the compiler recognizes them as valid intrinsic elements.
3. Action 1: Create a file at `src/types/ha-frontend.d.ts`. 
4. Action 2: Inside that file, use the following block to register the components:
   
   declare global {
     namespace JSX {
       interface IntrinsicElements {
         'ha-card': { header?: string; className?: string; children?: any; style?: any };
         'ha-icon': { icon?: string; className?: string; style?: any; slot?: string };
         'ha-settings-row': { className?: string; children?: any; slot?: string; narrow?: boolean };
         'ha-switch': { checked?: boolean; className?: string; disabled?: boolean };
         'ha-circular-progress': { active?: boolean; alt?: string; className?: string };
         'ha-tabs': { selected?: string; "attr-for-selected"?: string; scrollable?: boolean; children?: any };
         'paper-tab': { name?: string; children?: any };
       }
     }
   }
   export {};

5. Action 3: Verify the existence of `src/types/websocket.ts`. If it is missing, regenerate it with the WsCommand enum.
6. Action 4: Update `tsconfig.json`. Ensure `"include"` is set to `["src/**/*"]` to ensure the new `.d.ts` file is globally picked up by the compiler.
7. Always output the complete merged files and proactively apply optimizations.
8. Output the code for `src/types/ha-frontend.d.ts`, `src/types/websocket.ts`, and `tsconfig.json`.
</instructions>

Please apply these structural type fixes so the build can proceed.

### Specific Constraints

Ignore all errors not related to the meraki_ha integration. Follow Home Assistant Sentence Case standards.

--- a/custom_components/meraki_ha/www/src/types/ha-frontend.d.ts
+++ b/custom_components/meraki_ha/www/src/types/ha-frontend.d.ts
@@ -1,0 +1,16 @@
+import React from 'react';
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ha-card': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { header?: string }, HTMLElement>;
+      'ha-icon': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { icon?: string; slot?: string }, HTMLElement>;
+      'ha-settings-row': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { narrow?: boolean; slot?: string }, HTMLElement>;
+      'ha-switch': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { checked?: boolean; disabled?: boolean }, HTMLElement>;
+      'ha-circular-progress': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { active?: boolean; alt?: string }, HTMLElement>;
+      'ha-tabs': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { selected?: string; "attr-for-selected"?: string; scrollable?: boolean }, HTMLElement>;
+      'paper-tab': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & { name?: string }, HTMLElement>;
+    }
+  }
+}
+export {};

--- a/custom_components/meraki_ha/www/src/types/websocket.ts
+++ b/custom_components/meraki_ha/www/src/types/websocket.ts
@@ -1,0 +1,25 @@
+/**
+ * WebSocket Command Enum
+ */
+export enum WsCommand {
+  GET_CONFIG = 'meraki_ha/get_config',
+  SUBSCRIBE_MERAKI_DATA = 'meraki_ha/subscribe_meraki_data',
+  GET_CAMERA_STREAM_URL = 'meraki_ha/get_camera_stream_url',
+  GET_CAMERA_SNAPSHOT = 'meraki_ha/get_camera_snapshot',
+  GET_VERSION = 'meraki_ha/get_version',
+  GET_NETWORK_EVENTS = 'meraki_ha/get_network_events',
+  UPDATE_OPTIONS = 'meraki_ha/update_options',
+  UPDATE_ENABLED_NETWORKS = 'meraki_ha/update_enabled_networks',
+  TIMED_ACCESS_GET_KEYS = 'meraki_ha/timed_access/get_keys',
+  TIMED_ACCESS_GET_POLICIES = 'meraki_ha/timed_access/get_policies',
+  TIMED_ACCESS_CREATE = 'meraki_ha/timed_access/create',
+  TIMED_ACCESS_DELETE = 'meraki_ha/timed_access/delete',
+}
+
+/**
+ * WebSocket Message Payload Interface
+ */
+export interface WsMessagePayload {
+  type: WsCommand | string;
+  [key: string]: any;
+}

--- a/docs/REQUIREMENTS.md
+++ b/docs/REQUIREMENTS.md
@@ -139,6 +139,13 @@
 | :----------------------------------------------------- | :------- |
 | The Integration uses Voluptuous for schema validation. | Included |
 
+**Frontend Development
+
+| Requirement                                                                                                                                  | Status   |
+| :------------------------------------------------------------------------------------------------------------------------------------------- | :------- |
+| Home Assistant Web Components must be explicitly registered in the global JSX namespace (e.g., `src/types/ha-frontend.d.ts`) for TypeScript. | Included |
+| WebSocket commands used by the frontend must be centralized in a type-safe enum (e.g., `src/types/websocket.ts`).                            | Included |
+
 ## Key Learnings from Debugging
 
 | Requirement                                                                                | Status             |

